packages/babel-parser/src/parser/location.js

```diff
@@ -98,6 +98,11 @@ export const Errors = Object.freeze({
   MissingUnicodeEscape: "Expecting Unicode escape sequence \\uXXXX",
   MixingCoalesceWithLogical:
     "Nullish coalescing operator(??) requires parens when mixing with logical operators",
+  ModuleAttributeInvalidValue:
+    "Only string literals are allowed as module attributes values",
+  ModuleAttributesWithDuplicateKeys:
+    'Duplicate key "%0" is not allowed in module attributes',
+  ModuleAttributesWithoutType: 'Module attributes should include a "type" key',
```

Comment:
@littledan @xtuc We decided to add this error in the early stages of this PR. Is it still needed?

Reply:
To echo want I said on Slack. I think we should restrict the module attribute to the `type` only for now. It's easy enough to lift after and will avoid plugin author to already create custom attributes

Reply:
Updated the code to only allow as `type` as module attribute. The tests have now been updated to throw an error if anything is entered other than `type`.